programs/ssl/ssl_server2.c

```diff
@@ -2594,20 +2637,30 @@ int main( int argc, char *argv[] )
             }
         }
 
-        if ( mbedtls_pk_get_type( &pkey2 ) == MBEDTLS_PK_ECKEY ||
-             mbedtls_pk_get_type( &pkey2 ) == MBEDTLS_PK_RSA )
+        if( strcmp( opt.key2_opaque_alg1, DFL_KEY_OPAQUE_ALG ) == 0 )
```

Comment:
We're basically repeating ourselves from a few lines above, except with `s/key1/key2/`. Can't we extract that to a function, and while at it share it with the client?

Currently the client uses different defaults, but we could use `ecdsa-sign,ecdh` / `rsa-sign-pkcs1,rsa-sign-pss` everywhere just as well.